lms/djangoapps/ccx/views.py

```diff
@@ -338,6 +339,31 @@ def get_ccx_for_coach(course, coach):
     return None
 
 
+def get_date(ccx, node, date_type=None, parent_node=None):
+    """
+    This returns override or master date for section, subsection or a unit.
+
+    :param ccx: ccx instance
+    :param node: chapter, subsection or unit
+    :param date_type: start or due
+    :param parent_node: parent of node
+    :return: start or due date
+    """
+    date = get_override_for_ccx(ccx, node, date_type, None)
+    master_date = node.start if date_type == "start" else node.due
+    if date:
+        # Setting override date [start or due]
+        date = str(date)[:-9]
+    elif not parent_node and master_date:
+        # Get date from master course
+        date = str(master_date)[:-9]
+    elif parent_node:
+        # Set parent date (vertical has same dates as subsections)
+        date = get_date(ccx, parent_node, date_type)
```

Comment:
nit: just for being more readable and avoid confusion, I would write this as
`get_date(ccx, node=parent_node, date_type=date_type)`